src/AggregateFunctions/Streaming/CountedValueMap.h

```diff
@@ -72,48 +73,28 @@ class CountedValueMap
         return inserted;
     }
 
-    bool insert(T v)
+    /**
+    - If returnIfNewData is true, 
+        the function behaves returning true only if the element is new insert instead.
+    - If returnIfNewData is false (which is the default), 
+        the function behaves like the original insert,
+        returning true in all cases except when the capacity is full, and the new element is not desired.
+    */
```

Comment:
For a good interface, we should have a clear parameters and returns.
it's confusing

Reply:
is below works?
```cpp

    /**
    * Inserts an element into the map.
    * 
    * @param v The element to be inserted.
    * @param returnTrueIfNewData Controls the return behavior:
    *    - If true, returns true only if a new element is actually inserted.
    *    - If false (default), behaves like the standard insert: 
    *      returns true except when the capacity is full and the new element is not inserted.
    * 
    * @return True if the element is successfully processed (inserted or found), false otherwise.
    */
    bool insert(T v, bool returnTrueIfNewData = false)
    {
        if (atCapacity())
        {
            /// At capacity, this is an optimization
            /// fast ignore elements we don't want to maintain
            if (less(lastValue(), v))
                return false;
        }

        auto iter = m.find(v);
        if (iter != m.end())
        {
            ++iter->second;
            return !returnTrueIfNewData;
        }
        else
        {
            /// Didn't find v in the map
            [[maybe_unused]] auto [_, inserted] = m.emplace(arena->emplace(std::move(v)), 1);
            assert(inserted);

            eraseExtraElements();
            return true;
        }
    }

    /**
    * Erases an element from the map, if present.
    * 
    * @tparam TT The type of the element to be erased.
    * @param v The element to be erased.
    * @param returnTrueIfNewErased Controls the return behavior:
    *    - If true, returns true only if the element is found and actually erased (i.e., its count reaches zero).
    *    - If false (default), returns true if the element is found, regardless of whether it's actually erased.
    * 
    * @return True if the element is successfully processed based on the 'returnTrueIfNewErased' flag, false otherwise.
    */
    template <typename TT>
    bool erase(const TT & v, bool returnTrueIfNewErased = false) /// To enable heterogeneous erase
    {
        auto iter = m.find(v);
        if (iter != m.end())
        {
            --iter->second;
            if (iter->second == 0)
            {
                arena->free(iter->first);
                m.erase(iter);
                return true;
            }
            return !returnTrueIfNewErased;
        }
        return false;
    }

```